packages/amplication-plugin-api/src/plugin/plugin.resolver.ts

```diff
@@ -9,29 +9,55 @@ import { PluginService } from "./plugin.service";
 import { Public } from "../decorators/public.decorator";
 import { GraphQLError } from "graphql";
 import { PluginVersion } from "../pluginVersion/base/PluginVersion";
+import { PluginVersionService } from "../pluginVersion/pluginVersion.service";
+import { ProcessedPluginVersions } from "./plugin.types";
+import { PluginVersionFindManyArgs } from "../pluginVersion/base/PluginVersionFindManyArgs";
 
 @graphql.Resolver(() => Plugin)
 @common.UseGuards(GqlDefaultAuthGuard, gqlACGuard.GqlACGuard)
 export class PluginResolver extends PluginResolverBase {
   constructor(
     protected readonly service: PluginService,
     @nestAccessControl.InjectRolesBuilder()
-    protected readonly rolesBuilder: nestAccessControl.RolesBuilder
+    protected readonly rolesBuilder: nestAccessControl.RolesBuilder,
+    protected readonly pluginVersionService: PluginVersionService
   ) {
     super(service, rolesBuilder);
   }
 
   @Public()
-  @graphql.Query(() => [Plugin], { nullable: true })
-  async githubPlugins(): Promise<Plugin[]> {
+  @graphql.Mutation(() => [Plugin], { nullable: true })
+  async processPluginCatalog(): Promise<ProcessedPluginVersions[]> {
     try {
       const amplicationPlugins = await this.service.githubCatalogPlugins();
       if (
         Object.prototype.toString.call(amplicationPlugins) === "[object String]"
-      )
+      ) {
         throw amplicationPlugins;
+      }
 
-      return amplicationPlugins;
+      const npmPluginsVersions =
+        await this.pluginVersionService.npmPluginsVersions(amplicationPlugins);
+
+      const npmPluginsVersionsMap = npmPluginsVersions.reduce(
+        (acc, version) => {
+          const key = version.pluginId;
+          if (acc[key]) {
+            acc[key].push(version);
+          } else {
+            acc[key] = [version];
+          }
+          return acc;
+        },
+        {} as Record<string, PluginVersion[]>
```

Comment:
if map is required here, then why not use Map type instead of Record?

Reply:
moved to lodash groupBy